pkg/common/common_test.go

```diff
@@ -67,3 +71,192 @@ func TestFind(t *testing.T) {
 		t.Error("should not have found anything in the slice")
 	}
 }
+
+func TestFetchEnv(t *testing.T) {
+	t.Run("when env var exists & has a value different from the default value then return the env var value", func(t *testing.T) {
+		key := "LOG_LEVEL"
+		defaultVal := "info"
+		val := "debug"
+		os.Setenv(key, val)
+		defer os.Unsetenv(key)
+
+		result := FetchEnv(key, defaultVal)
+
+		if result != val {
+			t.Errorf("Expected %v, but got %v", val, result)
+		}
+	})
+	t.Run("when env var exists & has the same value as the default value then return the env var value", func(t *testing.T) {
+		key := "LOG_LEVEL"
+		defaultVal := "info"
+		val := "info"
+		os.Setenv(key, val)
+		defer os.Unsetenv(key)
+
+		result := FetchEnv(key, defaultVal)
+
+		if result != val {
+			t.Errorf("Expected %v, but got %v", val, result)
+		}
+	})
```

Comment:
I'd ditch this assertion, since it's not meaningful and would only add time to the whole battery of tests